### PR TITLE
Add electron example app

### PR DESCRIPTION
**Pull request by tortmayr**
#### What it does
- Add an electron example app in addition to the browser app. This allows use to also test the electron use case and makes it easier to discover electron specific issues.
- Adapt the build scripts to the same convention as used in the Theia repository.
   - yarn (install) only compiles the ts sources but does not build any example app
   - Example apps have to be built explicitly with `yarn browser build` or `yarn electron build`
- Rename glsp debug arugment to avoid clashes with nodes debug argument
- Update readmes and launch configs
- Update Jenkins file

#### How to test

Build the  the electron app 
eg:
`yarn && yarn electron build`
Then test the different connection scenarions
`yarn electron start`
`yarn electron start:ws`
`yarn electron start:debug`
`yarn electron start:ws:debug`
`yarn electron start:integrated`

#### Changelog

-   [] This PR should be mentioned in the changelog
-   [ ] This PR introduces a breaking change (if yes, provide more details below for the changelog and the migration guide)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,17 +23,9 @@
     "publish:latest": "lerna publish from-git --no-git-reset --no-git-tag-version --no-push",
     "publish:next": "lerna publish preminor --exact --canary --preid next --dist-tag next --no-git-reset --no-git-tag-version --no-push --ignore-scripts --yes",
     "publish:prepare": "lerna version --ignore-scripts --yes --no-push",
-    "start": "yarn browser start",
-    "start:debug": "yarn browser start:debug",
-    "start:integrated": "yarn browser start:integrated",
-    "start:ws": "yarn browser start:ws",
-    "start:ws:debug": "yarn browser start:ws:debug",
     "upgrade:next": "glsp updateNext",
     "watch": "concurrently --kill-others -n tsc,browser -c red,yellow \"tsc -b -w --preserveWatchOutput\" \"yarn -s browser watch\""
```
Comment (martin-fleck-at) [inline package.json:27]: nit: The `watch` only goes to the browser, should that also work for electron? If so we should either split it into `watch:browser` and `watch:electron` or simply add a `watch:electron`.

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Change looks good to me and I can open the Electron app and switch between Electron and Browser without issues. Thank you Tobias! I only noticed two things which may or may not be fixed as part of this PR:

- The toolbox styling is customized for Theia but I'm not sure we want all of them. If we do it is fine and we definitely want to use Theia color variables etc. But could you please double check whether all the Theia tool palette styling is as it should be.

- The grid does not work in the Electron case. It does work for the browser but I believe something does work and this stands out: `stroke="color-mix(in srgb, %23000000 10%, transparent)"`. This might be "fixed" with https://github.com/eclipse-glsp/glsp-client/pull/364 and some minor adaptations though, what do you think?

File: package.json (hunk 1 above)
Comment (tortmayr) [reply package.json]: Its now splitted

**Comment by tortmayr** [comment]
Thanks for the review @martin-fleck-at .
I have updated the PR and addressed the remaining issues.
- Toolbox styling has been adapted according to the latest client changes
- Grid styling now works for all theme types in browser and electron

I also sneaked in a minor improvement to the `createDiagramWidgetFactory` so that it can also be reused for configuring a widget factory for a subclass of GLSPDiagramWidget.

**Review by martin-fleck-at — APPROVED** [review]
Thank you for the quick update! I can confirm that the toolbox now looks nicer again and that the grid works in all themes. Great work, thank you, Tobias!

**Pull request merged**